examples/40_paper/2018_kdd_rijn_example.py

```diff
@@ -8,9 +8,116 @@
 ~~~~~~~~~~~
 
 | Hyperparameter importance across datasets
-| Jan van Rijn and Frank Hutter
+| Jan N. van Rijn and Frank Hutter
 | In *Proceedings of the 24th ACM SIGKDD International Conference on Knowledge Discovery & Data Mining*, 2018
 | Available at https://dl.acm.org/citation.cfm?id=3220058
-
-This is currently a placeholder.
 """
+import fanova
```

Comment:
ouch, that might be hard, as `pyrfr` is a direct requirement of this, which can not be installed through pip (as it depends on `swig`). Any ideas?

Reply:
You do `conda install gxx_linux-64 gcc_linux-64 swig` around line 40 in `ci_scripts/install.sh` (under the if clause `"$EXAMPLES" == "true"`). Then you can do `pip install pyrfr`.

Reply:
cool, thanks for your help! 

> pip install pyrfr

I don't think that would be necessary, should be handled in the `setup.py` with the fanova dependency, right?

Reply:
btw, we also need this under the doc test line? (currently the CI with doc_push seems to fail)

Also Appveyor does not like fanova (not sure if you can install `pyrfr` on windows machines)

Reply:
> I don't think that would be necessary, should be handled in the setup.py with the fanova dependency, right?

You're right.

> Also Appveyor does not like fanova

Interesting, I did not know that we try to install the dependencies for the examples under Windows, sorry for that. I also don't know why we do so, and I don't think it is necessary, so you could remove this test-wise. @PGijsbers do you have any idea why this is the case?

Reply:
Also, it seems that there's an issue in install.py as that the package actually isn't installed under examples any more, and therefore also not the dependencies (that's why fANOVA isn't there). It seems like you need to move the installation into the different if statements.

Reply:
apparently I needed to add the if-statement regarding  `$DOCPUSH` (assuming that this was the one that uses the examples, rather than coverage, correct me if I am wrong)

Regarding this statement: `conda install gxx_linux-64 gcc_linux-64 swig`, why doesn't there need to be a `--yes` in there? For some runs, the `--yes` seems not necessary, for others it is. (in particular, the one in the examples if-else is fine, whereas the one in docpush crashes otherwise)

Reply:
I think the answer to the last question is: the examples variable is nowhere used. 

I have added the --yes flag there as well, for consistency and competency. We should probably investigate.

Reply:
`swig` now installs, but `fanova` still doesn't. It currently crashes. I suspect this has to do with the missing "examples" flag.

Reply:
On installing example dependencies on Appveyor. It looks like I added the installation of example dependencies in commit c61064a. I'd say this is probably to keep the installed packages consistent with what was installed before the commit (as some notebook packages got installed). As we explicitly ignore unit tests for examples on Windows, we could forgo installing the example dependencies. 

But why are we ignoring examples on Windows? If we support Windows, shouldn't we test the examples (or at least those that only rely on straight forward installable packages)? I think it's reasonable for a Windows user to install `openml-python` with current examples and run them - not get some error because installing `pyrfr` can be problematic out of the box. 

Of course we could set up our Appveyor scripts to test these examples with more difficult dependencies. However, end-users will then have to resolve any installation issues too, which I think is messy (and probably will create additional issues on our repo for users being unable to install e.g. `pyrfr`). For that reason I would propose something along the lines of dividing examples in two: examples which install easily out of the box cross-platform, and examples which don't. The messiness of having two example categories is in my opinion still cleaner than the messiness of effectively being unable to run `pip install openml-python[examples]` on Windows.

Reply:
> doesn't there need to be a `--yes` in there

Thinking about this again, I'd also say you need a `--yes` there.

> apparently I needed to add the if-statement regarding $DOCPUSH

I just realized that, too. Sorry for not noticing this earlier. You can copy the code from my PR #806 to make it clearer.

> not get some error because installing `pyrfr` can be problematic out of the box.

I don't know if that will be possible at all. Should we use two categories `[examples-windows]` and `[examples-unix]` to differentiate here? Then we'd only have to add a note to the fanova example to highlight that it won't work under windows.

Reply:
> I don't know if that will be possible at all. Should we use two categories [examples-windows] and [examples-unix] to differentiate here? Then we'd only have to add a note to the fanova example to highlight that it won't work under windows.

Yes, that is exactly what I had in mind. I'm all for cross-platform support, but I do not think it warrants the time to include the fanova in Appveyor CI or to try and support Windows users from running the fanova experiment.  Having a clean demarkation of which examples are not supported on Windows (for now just fanova) and making sure it's easy the install dependencies for just the examples that work on your platform seems like the way to go to me.